sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -4,3 +4,96 @@
 
 //! Useful modules for OpenTitanTool application development.
 pub mod command;
+pub mod conf;
+
+use crate::io::gpio::GpioPin;
+use crate::io::spi::Target;
+use crate::io::uart::Uart;
+
+use anyhow::Result;
+use std::collections::HashMap;
+use std::rc::Rc;
+
+// This is the structure to be passed to each Command implementation,
+// replacing the "bare" Transport argument.  The fields other than
+// transport will have been computed from a number ConfigurationFiles.
+pub struct TransportWrapper {
+    transport: std::cell::RefCell<Box<dyn crate::transport::Transport>>,
+    pub pin_map: HashMap<String, String>,
+    pub uart_map: HashMap<String, String>,
+    pub spi_map: HashMap<String, String>,
+    pub flash_map: HashMap<String, conf::FlashConfiguration>,
+}
+
+impl TransportWrapper {
+    pub fn new(transport: Box<dyn crate::transport::Transport>) -> Self {
+        Self {
+            transport: std::cell::RefCell::new(transport),
+            pin_map: HashMap::new(),
+            uart_map: HashMap::new(),
+            spi_map: HashMap::new(),
+            flash_map: HashMap::new(),
+        }
+    }
+
+    /// Returns a `Capabilities` object to check the capabilities of this
+    /// transport object.
+    pub fn capabilities(&self) -> crate::transport::Capabilities {
+        self.transport.borrow().capabilities()
+    }
+
+    /// Returns a SPI [`Target`] implementation.
+    pub fn spi(&self, name: &str) -> Result<Rc<dyn Target>> {
+        self.transport.borrow().spi(Self::map_name(&self.spi_map, name).as_str())
+    }
+
+    /// Returns a [`Uart`] implementation.
+    pub fn uart(&self, name: &str) -> Result<Rc<dyn Uart>> {
+        self.transport.borrow().uart(Self::map_name(&self.uart_map, name).as_str())
+    }
+
+    /// Returns a [`GpioPin`] implementation.
+    pub fn gpio_pin(&self, name: &str) -> Result<Rc<dyn GpioPin>> {
+        self.transport.borrow().gpio_pin(Self::map_name(&self.pin_map, name).as_str())
+    }
+
+    /// Given an pin/uart/spi port name, if the name is a known alias,
+    /// return the underlying name/number, otherwise return the string
+    /// as is.
+    fn map_name(
+        map: &HashMap<String, String>,
+        name: &str
+    ) -> String {
+        let name = name.to_uppercase();
+        // TODO(jbk): Support multi-level aliasing, either by
+        // flattening after parsing all files, or by repeated lookup
+        // here.
+        match map.get(&name) {
+            Some(alias) => alias.clone(),
+            None => name.to_string()
+        }
+    }
+
+    pub fn add_configuration_file(&mut self, file: conf::ConfigurationFile) -> Result<()> {
+        // Merge content of configuration file into pin_map and other
+        // members.
+        for pin_conf in file.pins {
```

Comment:
You could do `file.pins.into_iter().filter_map(|x| x)`. Seeing as `x` is already an Option, filter_map Just Works.